### PR TITLE
cli: add cli support for CIL

### PR DESCRIPTION
**Pull request by manorit2001**

### CODE REVIEW

**Review by parth-07 — APPROVED** [review]
LGTM.

File: examples/main.cpp [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        case methods::flip:
+        {
+            function_parameters::flip* args = (function_parameters::flip*)
+                                                  handler->data;
+
+            ret = IS_INVALID(args->axis);
+            if (ret)
+                goto err;
+
+            CIL::flipImage(img, static_cast<CIL::Axis>(args->axis - 1));
```
Comment (parth-07) [inline examples/main.cpp:202]: Should we add a check that `args->axis` has value 1 or 2?

**Comment by parth-07** [comment]
Why the entire CLI code is in example files? (**rude comment**).

**Comment by manorit2001** [comment]
> Why the entire CLI code is in example files? (**rude comment**).

Coz the initial design you chose didn't have any better place then it, wanna shift it to `lib` folder? I AM IN

**Comment by parth-07** [comment]
> Coz the initial design you chose

You chose? I remember "We" chosing the initial design !!

> wanna shift it to lib folder?

How about a separate folder for CLI such as `/tools/`?

**Comment by manorit2001** [comment]
> > Coz the initial design you chose
> 
> You chose? I remember "We" chosing the initial design !!

* ignoring this comment *

> > wanna shift it to lib folder?
> 
> How about a separate folder for CLI such as `/tools/`?

Hm, I am in. But you know what would be more cool? `lib` folder

File: examples/main.cpp (hunk 1 above)
Comment (manorit2001) [reply examples/main.cpp]: Fixed

**Comment by manorit2001** [comment]
@parth-07 could you fix the CMakeLists.txt inside the tools folder? I wasn't able to make it work like the examples folder one did

**Comment by parth-07** [comment]
> @parth-07 could you fix the CMakeLists.txt inside the tools folder? I wasn't able to make it work like the examples folder one did
> 
> The folder name in the install folder where the binary comes should be `bin`

Done, can you please try with CMake in your system?

```
cmake -G Ninja ../cli -DCMAKE_INSTALL_PREFIX=../inst -DCIL_BUILD_CLI=On
```

`ninja install` will install cli in `bin` directory of the install folder.

**Pull request merged**